packages/vite/src/node/config.ts

```diff
@@ -487,9 +501,20 @@ export async function resolveConfig(
         preserveSymlinks: config.resolve?.preserveSymlinks,
         ...config.optimizeDeps?.esbuildOptions
       }
-    }
+    },
+    worker: resolvedWorkerOptions
   }
 
+  // flat config.worker.plugin
+  const [workerPrePlugins, workerNormalPlugins, workerPostPlugins] =
+    sortUserPlugins(config.worker?.plugins as Plugin[])
+
+  resolved.worker.plugins = await resolvePlugins(
+    { ...resolved },
```

Comment:
We can save `...resolved` as `workerConfig`, so we can also call resolvedConfig for worker plugins. See later in the code: 
```
  await Promise.all(userPlugins.map((p) => p.configResolved?.(resolved)))
```